packages/loader/lib/index.js

```diff
@@ -35,7 +35,10 @@ const cache = new WeakMap()
 export function loader(value, callback) {
   /** @type {Defaults} */
   const defaults = this.sourceMap ? {SourceMapGenerator} : {}
-  const options = /** @type {CompileOptions} */ (this.getOptions())
+  const options = {
+    development: this.mode === 'development',
+    .../** @type {CompileOptions} */ (this.getOptions())
+  }
```

Comment:
From reading [the docs](https://webpack.js.org/configuration/mode/), it seems like `mode: development` sets `process.env.NODE_ENV`? If that’s the case, shouldn’t we already be picking up on this?

Reply:
No, it uses `DefinePlugin` to define `process.env.NODE_ENV` in the bundle. It doesn’t set the actual environment variable, so the loader doesn’t pick it up.

I juse tried it using the following setup:

```js
// webpack.config.js
export default {
  module: {
    rules: [
      { use: './loader.cjs' }
    ]
  }
}
```

```js
// loader.cjs
module.exports = (content) => {
  console.log(process.env.NODE_ENV)
  return 'console.log("loader");' + content
}
```

```js
// index.js
console.log(process.env.NODE_ENV)
```

Now building yields:

```
$ webpack build --mode production
process.env.NODE_ENV undefined
asset main.js 49 bytes [emitted] [minimized] (name: main)
./src/index.js 33 bytes [built] [code generated]
webpack 5.75.0 compiled successfully in 153 ms
```

And this is the build output:

```js
(()=>{"use strict";console.log("production")})();
```